[BE] Fix bugs

MihaiDruta opened the pull request:
[BE] Fix bugs
https://trello.com/c/NPEi8KKJ/95-be-fix-bugs

Fixed bugs:
Feedback:
- User is now able to give feedback only once to a specific user involved on an activity

Pagination:
- If the page is negative a HttpNotFoundException is thrown
- If the page exceeds the maximum number of pages available a HttpNotFoundException is thrown

Commit:
[BE] Fix bugs
https://trello.com/c/NPEi8KKJ/95-be-fix-bugs

Fixed bugs:
Feedback:
- User is now able to give feedback only once to a specific user involved on an activity
Pagination:
- If the page is negative a HttpNotFoundException is thrown
- If the page exceeds the maximum number of pages available a HttpNotFoundException is thrown

SpartakusMd commented on src/Filters/PaginatorPageFieldValidator.php:
```diff
@@ -0,0 +1,11 @@
+
+namespace App\Filters;
+
+class PaginatorPageFieldValidator
```
Maybe rename class to `PaginatorValidator` and the method to `isPageNumberValid`?

SpartakusMd commented on src/Repository/FeedbackRepository.php:
```diff
@@ -88,4 +89,12 @@ public function getUserFeedbackPaginated(
         return $query;
     }
+
+    public function canUserGiveFeedback(Activity $activity, User $userFrom, User $userTo): bool
```
This method doesn't implement all the checks necessary, so it's name should be changed to `hasUserGivenFeedback()`

SpartakusMd commented on src/Handlers/UserHandler.php:
```diff
@@ -54,12 +63,17 @@ public function getUserListPaginated(
+        $numPages = (int)ceil($numResults / $userListPagination->pageSize);
+        if (!$this->parameterValidator->isParameterValid($numPages, $userListPagination->currentPage)) {
+            throw new NotFoundHttpException();
+        }
```
The block `$context = ...` and `$json = $this->serializer...` should be after this validation. Check all the controllers.

SpartakusMd commented on src/Handlers/UserHandler.php:
```diff
@@ -54,12 +63,17 @@ public function getUserListPaginated(
             $userListPagination->pageSize = $numResults;
```
Add this feature to all the paginated lists.

Commit:
[BE] Fix bugs
https://trello.com/c/NPEi8KKJ/95-be-fix-bugs

1. Renamed pagiantion field validator class and it's method to PaginatorValidator.php, hasUserGivenFeedback()
2. We can now see all entries by setting up page size to -1 on all paginated lists

SpartakusMd commented on src/Repository/ActivityRepository.php:
```diff
@@ -122,10 +122,13 @@ public function getPaginatedActivities(
     ): Query {
         $queryBuilder = $this->getAvailableActivities($activityListSort, $activityListFilter, $user);
-
+        if ($activityListPagination->pageSize === -1) {
```
We can't have anymore `pageSize = -1` as we have `PaginatorValidator` :) Check all repozitories.

Commit:
[BE] Fix bugs
https://trello.com/c/NPEi8KKJ/95-be-fix-bugs

Added validation for pageSize also

SpartakusMd commented:
@MihaiDruta please fix conflicts.
